File: src/Core/ViewModels/MessageViewModel.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Core.ViewModels
{
    public sealed class MessageViewModel
    {
        public bool Owner { get; set; }
        public DateTime DateSent { get; set; }

        [Required]
        public string Context { get; set; }
    }
}

File: src/AngularApp/ClientApp/src/app/models/MessageViewModel.ts
export class MessageViewModel {
  constructor(context: string, dateSent: string, owner: boolean) {
    this.context = context;
    this.dateSent = dateSent;
    this.owner = owner;
  }

  public context: string;
  
  public dateSent: string;
  
  public owner: boolean;
}
